go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -496,8 +503,8 @@ func (n *OvnNode) initSharedGateway(subnets []*net.IPNet, gwNextHop net.IP, gwIn
 		ChassisID:      chassisID,
 		InterfaceID:    ifaceID,
 		MACAddress:     macAddress,
-		IPAddresses:    []*net.IPNet{ipAddress},
-		NextHops:       []net.IP{gwNextHop},
+		IPAddresses:    ips,
```

Comment:
`IPAddresses` in the `L3GatewayConfig` is not just the raw list of all available node IPs; if this is a single-stack cluster (ie, `len(subnets) == 1`) then `IPAddresses` needs to have exactly 1 address, of the correct family. If this is a dual-stack cluster, it needs to have exactly 2 addresses, one of each family. (They don't necessarily need to be in the same order as `subnets`.)

Reply:
Updated getNetworkInterfaceIPAddresses() to only add one address from each family, if enabled. I add the first address found in a family and skip any others found (not sure there would ever be more than one per family, but now it is enforced).
Done.